core/src/test/java/com/graphhopper/storage/MMapDataAccess_DiffblueTest.java

```diff
@@ -0,0 +1,310 @@
+/**
+ *
+ *
+ * Copyright 2018 Diffblue Limited
+ *
+ * Diffblue Limited licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.graphhopper.storage;
+
+import static org.mockito.AdditionalMatchers.or;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Matchers.isNull;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+import com.diffblue.deeptestutils.Reflector;
+import com.graphhopper.storage.AbstractDataAccess;
+import com.graphhopper.storage.DAType.MemRef;
+import com.graphhopper.storage.DAType;
+import com.graphhopper.storage.MMapDataAccess;
+import com.graphhopper.util.BitUtilBig;
+import com.graphhopper.util.Helper;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.File;
+import java.io.RandomAccessFile;
+import java.lang.reflect.Array;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+
+@PowerMockIgnore({"javax.management.*"})
+@RunWith(PowerMockRunner.class)
+public class MMapDataAccess_DiffblueTest {
+
```

Comment:
Same as above for this test class - I'm not sure, lots of reflections but it is for a private method. @peterschrammel @petersear and @James-Adam what do you think.

Reply:
👎  reflective construction is no good here

Reply:
Can you create a bug ticket for this as a challenge to instantiate this non-reflectively?